### PR TITLE
Update ticket log view

### PR DESCRIPTION
**Pull request by kalebdavis**
Uses an accordion view to display the ticket logs in a less cluttered way.

screen shot 2017-05-07 at 4 56 16 pm

### CODE REVIEW

File: src/routes/Group/components/TicketView.js [hunk 1]
```diff
@@ -23,37 +24,102 @@ class TicketView extends React.Component {
...
-      rows.push(<Table.Cell key={'date_'+id}>{String(new Date(action.timestamp))}</Table.Cell>);
-      rows.push(<Table.Cell key={'user_'+id}>{action.user ? action.user.name : ''}</Table.Cell>);
-      rows.push(<Table.Cell key={'device_'+id}>{action.device ? action.device.type : ''}</Table.Cell>);
+
+    const action = ticket.actions[index];
+    let summary;
+    switch(action.actionTaken) {
+      case 'CREATED':
+        summary = `${this.capitalize(ticket.metadata.title)} created.`;
+        events.push(this.eventHelper(index, action.timestamp, summary));
```
Comment (jarrydlee) [inline src/routes/Group/components/TicketView.js]: You're doing this 3 times, you could just pull it out

File: src/routes/Group/components/TicketView.js [hunk 2]
```diff
@@ -23,37 +24,102 @@ class TicketView extends React.Component {
...
-      </Table.Row>
+      <Feed key={index}>
+        {events}
+      </Feed>
     );
   }
 
+  renderLogs() {
+    if (!this.props.tickets) {
+      return <div><Header as="h3">No Ticket History</Header></div>;
```
Comment (jarrydlee) [inline src/routes/Group/components/TicketView.js]: Do you need a `<div></div>` here?

File: src/routes/Group/components/TicketView.js [hunk 3]
```diff
@@ -23,37 +23,102 @@ class TicketView extends React.Component {
...
+        activeIndex={this.state.activeIndex}
+        panels={panels}
+        onTitleClick={this.handleAccordionTitleClick}
+        fluid
+        styled
+      />
+    )
+  }
+
+  capitalize(str) {
```
Comment (kwm4385) [inline src/routes/Group/components/TicketView.js]: There's a lodash for this https://lodash.com/docs/#capitalize

File: src/routes/Group/components/TicketView.js [hunk 4]
```diff
@@ -23,37 +23,102 @@ class TicketView extends React.Component {
...
+
+  formatDate(date) {
+    return moment(date).format('MMMM D YYYY, h:mm:ss a');
+  }
+
+  eventHelper(index, timestamp, summary) {
+    return (
+      <Feed.Event key={index}>
+        <Feed.Content>
+          <Feed.Date content={String(this.formatDate(timestamp))} />
```
Comment (kwm4385) [inline src/routes/Group/components/TicketView.js]: Doesn't moment.format() already return a string?

File: src/routes/Group/components/TicketView.js [hunk 5]
```diff
@@ -23,37 +23,102 @@ class TicketView extends React.Component {
...
+
   ticketRow(ticket, index){
-    const title = (ticket.metadata.title) ? ticket.metadata.title : 'No name';
-    const rows = [];
+    const events = []
     const id = 'ticket_'+ticket._id+'_'+index;
-    if (index == 0) {
-      rows.push(<Table.Cell key={'title_'+id} rowSpan={ticket.actions.length}>{title}</Table.Cell>);
+    if (!ticket.actions) {
+      return ('No ticket actions');
```
Comment (kwm4385) [inline src/routes/Group/components/TicketView.js]: In rendering you generally always want to return a component. I'd just wrap this in a <span>

**Pull request merged**